File: fetch.js
const fs = require('fs');

const files = [
    "mar.jpeg",
    "csharpintro.png",
    "scharpinter.png",
    "pyton%20intro.png",
    "html.png",
    "css.png",
    "introblazor.PNG",
    "firstapp.PNG",
    "todolist.PNG"
];

const baseUrl = "https://raw.githubusercontent.com/Marouan-el-yassini/Marwan-El-Yassini-Website/f10590f/";

async function downloadFiles() {
    for (const file of files) {
        try {
            console.log(`Downloading ${file}...`);
            const response = await fetch(baseUrl + file);
            if (!response.ok) throw new Error(`Status ${response.status}`);
            
            const buffer = await response.arrayBuffer();
            const outName = decodeURIComponent(file);
            fs.writeFileSync(outName, Buffer.from(buffer));
            console.log(`Saved ${outName}`);
        } catch (err) {
            console.error(`Error downloading ${file}:`, err.message);
        }
    }
    console.log("All done.");
}

downloadFiles();

File: fetch.py
import urllib.request
import threading
import sys

files = [
    "mar.jpeg",
    "csharpintro.png",
    "scharpinter.png",
    "pyton%20intro.png",
    "html.png",
    "css.png",
    "introblazor.PNG",
    "firstapp.PNG",
    "todolist.PNG"
]

base_url = "https://raw.githubusercontent.com/Marouan-el-yassini/Marwan-El-Yassini-Website/f10590f/"

def download(file):
    try:
        url = base_url + file
        out_file = file.replace("%20", " ")
        print(f"Downloading {out_file}...")
        urllib.request.urlretrieve(url, out_file)
        print(f"Success: {out_file}")
    except Exception as e:
        print(f"Failed {file}: {e}")

threads = []
for f in files:
    t = threading.Thread(target=download, args=(f,))
    threads.append(t)
    t.start()

for t in threads:
    t.join()

print("All downloads complete.")
